lib/collection/src/grouping/group_by.rs

```diff
@@ -224,21 +249,68 @@ impl From<RecommendGroupsRequestInternal> for GroupRequest {
     }
 }
 
+impl From<QueryGroupsRequestInternal> for GroupRequest {
+    fn from(request: QueryGroupsRequestInternal) -> Self {
+        let QueryGroupsRequestInternal {
+            prefetch,
+            query,
+            using,
+            filter,
+            params,
+            score_threshold,
+            with_vector,
+            with_payload,
+            group_request:
+                BaseGroupRequest {
+                    group_by,
+                    group_size,
+                    limit,
+                    with_lookup: with_lookup_interface,
+                },
+        } = request;
+
+        let collection_query_request = CollectionQueryRequest {
+            prefetch: prefetch.into_iter().flatten().map(From::from).collect(),
+            query: query.map(From::from),
+            using: using.unwrap_or(DEFAULT_VECTOR_NAME.to_string()),
+            filter,
+            score_threshold,
+            limit: limit as usize,
+            offset: 0,
+            params,
+            with_vector: with_vector.unwrap_or(CollectionQueryRequest::DEFAULT_WITH_VECTOR),
+            with_payload: with_payload.unwrap_or(CollectionQueryRequest::DEFAULT_WITH_PAYLOAD),
+            lookup_from: None,
+        };
+
+        GroupRequest {
+            source: SourceRequest::Query(collection_query_request),
+            group_by,
+            group_size: group_size as usize,
+            limit: limit as usize,
+            with_lookup: with_lookup_interface.map(Into::into),
+        }
+    }
+}
+
 /// Uses the request to fill up groups of points.
 pub async fn group_by(
-    request: CoreGroupRequest,
+    request: QueryGroupRequest,
     collection: &Collection,
     read_consistency: Option<ReadConsistency>,
     shard_selection: ShardSelectorInternal,
     timeout: Option<Duration>,
 ) -> CollectionResult<Vec<PointGroup>> {
-    let score_ordering = if request.source.query.is_distance_scored() {
-        let vector_name = request.source.query.get_vector_name();
-        let collection_params = collection.collection_config.read().await;
-        let distance = collection_params.params.get_distance(vector_name)?;
-        distance.distance_order()
-    } else {
-        Order::LargeBetter
+    let mut score_ordering = Order::LargeBetter;
+
+    if let Some(query) = &request.source.query {
+        if query.is_distance_scored() {
+            if let Some(vector_name) = query.get_vector_name() {
+                let collection_params = collection.collection_config.read().await;
+                let distance = collection_params.params.get_distance(vector_name)?;
+                score_ordering = distance.distance_order()
+            }
+        }
     };
```

Comment:
We also need to consider `order_by` direction. There is a [`ScoringQuery::order`](https://github.com/qdrant/qdrant/blob/dev/lib/collection/src/operations/universal_query/shard_query.rs#L83) function which can be used in here :)